src/App.tsx

```diff
@@ -1,222 +1,194 @@
+    = useState<number[]>([]);
+
+  const showErrorMessage = useCallback((error: string) => {
+    setErrorMessage(error);
+
+    setTimeout(() => setErrorMessage(''), 3000);
+  }, []);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    if (user) {
+      getTodos(user.id)
+        .then(todo => (
+          setTodos(todo)))
+        .catch(() => showErrorMessage('Failed request for todos'));
     }
   }, []);
 
+  const addTodo = async (todoData: Omit<Todo, 'id'>) => {
+    try {
+      const tempTodo = {
+        ...todoData,
+        id: 0,
+      };
+
+      setTemporaryTodo(tempTodo);
+
+      const newTodo = await creatTodo(todoData);
+
+      setTodos(currentTodoArray => [...currentTodoArray, newTodo]);
+      setTemporaryTodo(null);
+    } catch {
+      showErrorMessage('Unable to add a todo');
+      setTemporaryTodo(null);
+    }
+  };
+
+  const deletedTodo = async (todoId: number) => {
+    try {
+      setDeletingTodoIds(prevList => [...prevList, todoId]);
+
+      const todoToDelete = await deletingTodo(todoId);
+
+      setTodos(oldTodos => oldTodos.filter(todo => todo.id !== todoId));
+
+      setDeletingTodoIds(prevList => (
+        prevList.filter(id => id !== todoId)
+      ));
+
+      return todoToDelete;
+    } catch {
+      setDeletingTodoIds(prevList => (
+        prevList.filter(id => id !== todoId)
+      ));
+      showErrorMessage('Unable to delete a todo');
+
+      return false;
+    }
+  };
+
+  const filteredTodosByStatus = useMemo(() => (
+    filteredTodos(todoFilterComplete, todos)),
+  [todoFilterComplete, todos]);
+
+  const uncompletedTodos = useMemo(
+    () => todos.filter(todo => !todo.completed), [todos],
+  );
+
+  const completedTodos = useMemo(
+    () => todos.filter(todo => todo.completed), [todos],
+  );
+
+  const visibleListContent = todos.length !== 0 || temporaryTodo;
+
+  const getCompletedTodos = useCallback(() => {
+    const filtredTodos = getTodoCompletedId(todos);
+
+    return filtredTodos.forEach(id => deletedTodo(id));
+  }, [todos]);
+
+  const updatingTodo = async (todoForUpdating: Todo) => {
+    try {
+      setUpdatingTodoIds(todoprev => [...todoprev, todoForUpdating.id]);
+
+      const {
+        ...valuesToUpdate
+      } = todoForUpdating;
+
+      const updatedTodo = await updatingTodoOnServer(valuesToUpdate);
+
+      setTodos((currentTodo) => currentTodo.map(todo => (
+        todo.id === updatedTodo.id
+          ? updatedTodo
+          : todo
+      )));
+    } catch {
+      showErrorMessage('Unable to update a todo');
+    } finally {
+      setUpdatingTodoIds([]);
+    }
+  };
+
+  const toggleAllTodos = useCallback(() => {
+    todos.forEach(todo => {
+      if (!completedTodos.includes(todo)) {
+        updatingTodo({
+          ...todo,
+          completed: !todo.completed,
+        });
+      } else if (completedTodos.length === todos.length) {
```

Comment:
You could create the variable were you store two options

`!completedTodos.includes(todo)` and `completedTodos.length === todos.length` and use one `if`